CMakeLists.txt

```diff
@@ -31,8 +31,17 @@ endif()
 
 cmake_dependent_option(ARBORX_ENABLE_ONEDPL "Enable oneDPL support" ON "Kokkos_ENABLE_SYCL" OFF)
 if(Kokkos_ENABLE_SYCL AND ARBORX_ENABLE_ONEDPL)
-  find_package(oneDPL REQUIRED)
-  target_link_libraries(ArborX INTERFACE oneDPL)
+  include(CheckIncludeFileCXX)
+  check_include_file_cxx(oneapi/dpl/execution ARBORX_COMPILER_HAS_ONEDPL_EXECUTION_HEADER)
+  check_include_file_cxx(oneapi/dpl/algorithm ARBORX_COMPILER_HAS_ONEDPL_ALGORITHM_HEADER)
+  if (!ARBORX_COMPILER_HAS_ONEDPL_EXECUTION_HEADER OR !ARBORX_COMPILER_HAS_ONEDPL_ALGORITHM_HEADER)
+    find_package(oneDPL REQUIRED)
+    target_link_libraries(ArborX INTERFACE oneDPL)
+  endif()
+  # libstdc++-9 and libstdc++-10 using legacy TBB that is incompatible with oneTBB, see
+  # https://stackoverflow.com/a/67924408
+  target_compile_definitions(ArborX INTERFACE _GLIBCXX_USE_TBB_PAR_BACKEND=0)
+  target_compile_definitions(ArborX INTERFACE PSTL_USE_PARALLEL_POLICIES=0)
```

Comment:
This is debatable but pretty much everyone using `oneDPL` (with GCC-9 or GCC-10 headers) needs to do this one way or the other.

Reply:
Obviously, the compiler we use for CI doesn't need it but I need it on the testbeds. It seems to be a toolchain issue depending on the standard library implementation used.
In any case, I dropped it for now. We can discuss adding it elsewhere.